### PR TITLE
[Bitcoind] Add rpc bind and allow for docker setup

### PR DESCRIPTION
**Pull request by MontejoJorge**
## Summary by CodeRabbit

- **New Features**
  - Updated `tipi_version` from 4 to 5 in the `bitcoind` app for improved functionality.
  - Enabled RPC server access from any IP address and bound RPC server to all network interfaces in the `bitcoind` configuration.

### CODE REVIEW

File: apps/bitcoind/docker-compose.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ services:
       -maxmempool=${BITCOIND_MAX_MEMPOOL:-300}
       -listen=${BITCOIND_LISTEN:-0}
       -server=${BITCOIND_SERVER:-1}
+      -rpcallowip=0.0.0.0/0
+      -rpcbind=0.0.0.0
```
[the remark this answers is not in the document]
Comment (MontejoJorge) [reply apps/bitcoind/docker-compose.yml:22]: Necesary config for docker

**Comment by DireMunchkin** [comment]
Hey, thanks for taking a look @MontejoJorge, still some issue though.

```
electrs-ui  |
electrs-ui  | > electrs-ui@0.9.2 dev:backend /app
electrs-ui  | > pnpm --filter backend run start
electrs-ui  |
electrs-ui  |
electrs-ui  | > backend@0.1.12 start /app/apps/backend
electrs-ui  | > node ./bin/www
electrs-ui  |
electrs-ui  | Thu, 13 Jun 2024 07:21:08 GMT morgan deprecated morgan(options): use morgan("default", options) instead at app.js:30:9
electrs-ui  | Thu, 13 Jun 2024 07:21:08 GMT morgan deprecated default format: use combined format at app.js:30:9
electrs-ui  | Listening on port 3006
electrs     | Starting electrs 0.10.2 on x86_64 linux with Config { network: Bitcoin, db_path: "/data/db/bitcoin", db_log_dir: None, daemon_dir: "/root/.bitcoin", daemon_auth: CookieFile("/data/.bitcoin/.cookie"), daemon_rpc_addr: 192.168.1.170:8332, daemon_p2p_addr: 192.168.1.170:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs 0.10.2 (Electrum Rust Server)!", signet_magic: f9beb4d9, args: [] }
electrs     | [2024-06-13T07:21:07.937Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
electrs     | [2024-06-13T07:21:07.937Z INFO  electrs::server] serving Electrum RPC on 0.0.0.0:50001
electrs     | [2024-06-13T07:21:07.978Z INFO  electrs::db] "/data/db/bitcoin": 429 SST files, 0.0004367 GB, 0.000000021 Grows
electrs     | [2024-06-13T07:21:08.005Z WARN  electrs::p2p] failed to shutdown p2p connection: Transport endpoint is not connected (os error 107)
electrs     | [2024-06-13T07:21:08.005Z INFO  electrs::db] closing DB at /data/db/bitcoin
electrs     | Error: electrs failed
electrs     |
electrs     | Caused by:
electrs     |     receiving on an empty and disconnected channel
electrs     | Starting electrs 0.10.2 on x86_64 linux with Config { network: Bitcoin, db_path: "/data/db/bitcoin", db_log_dir: None, daemon_dir: "/root/.bitcoin", daemon_auth: CookieFile("/data/.bitcoin/.cookie"), daemon_rpc_addr: 192.168.1.170:8332, daemon_p2p_addr: 192.168.1.170:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs 0.10.2 (Electrum Rust Server)!", signet_magic: f9beb4d9, args: [] }
electrs     | [2024-06-13T07:21:08.651Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
electrs     | [2024-06-13T07:21:08.651Z INFO  electrs::server] serving Electrum RPC on 0.0.0.0:50001
electrs     | [2024-06-13T07:21:08.685Z INFO  electrs::db] "/data/db/bitcoin": 430 SST files, 0.000437718 GB, 0.000000021 Grows
electrs     | [2024-06-13T07:21:08.688Z WARN  electrs::thread] p2p_recv thread failed: failed to recv a message from peer: IO error
electrs     | [2024-06-13T07:21:08.688Z WARN  electrs::p2p] failed to shutdown p2p connection: Transport endpoint is not connected (os error 107)
electrs     | [2024-06-13T07:21:08.688Z INFO  electrs::db] closing DB at /data/db/bitcoin
```

**Comment by DireMunchkin** [comment]
Yes that was it, now it started indexing 🙌

Maybe enable that as default as well like with RPC on / RPC bind?

I'll just check that I can point Sparrow Wallet to Electrs after I finish syncing this and I think we're done.

**Comment by MontejoJorge** [comment]
Thanks you for your help @DireMunchkin
I'll appreciate  some feedback on #3434 if you have time ^^

**Comment by DireMunchkin** [comment]
> Thanks you for your help @DireMunchkin I'll appreciate some feedback on #3434 if you have time ^^

No, thank you! Since you put in the work to port these now I can run just Tipi instead of Tipi and Umbrel. 😁

I'll have a look at your Mempool PR next.

**Comment by DireMunchkin** [comment]
I was able to make use of Electrs from my wallet, looks all good to me.

@meienberger feel free to merge

**Comment by meienberger** [comment]
Merged!

**Pull request merged**